frontend/src/Pages/ProgramOverviewDashboard.tsx

```diff
@@ -243,10 +248,15 @@ export default function ProgramOverview() {
                         </button>
                     ) : (
                         <button
-                            className="button flex items-center"
+                            className={`button flex items-center ${isAddClassDisabled ? 'opacity-50 cursor-not-allowed' : ''}`}
                             onClick={() =>
                                 navigate(`/programs/${id}/classes/new`)
                             }
+                            disabled={
+                                !program.facilities.some(
+                                    (facility) => facility.id === userFacilityId
+                                )
```

Comment:
@corypride Rather than defining it again, you can reuse the same expression value that you added to handle conditional styling.